File: 코딩 기초 트레이닝/정사각형으로 만들기-p181830/정사각형으로 만들기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181830
function solution(arr) {
    if(arr.length > arr[0].length) {
        let add = arr.length - arr[0].length;

        for(let i = 0 ; i < arr.length ; i++) {
            let a = new Array(add).fill(0);
            arr[i].push(...a);
        }
        // 행 < 열
    } else if(arr.length < arr[0].length) {
        // 열 < 행
        let add = arr[0].length - arr.length;

        for(let i = 0; i < add ; i++ ) {
            let a = new Array(arr[0].length).fill(0);
            arr.push(a);
        }
    }

    return arr;
}

File: 코딩 기초 트레이닝/정사각형으로 만들기-p181830/정사각형으로 만들기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181830

def solution(arr):
    if len(arr) > len(arr[0]):
        LEN = len(arr) - len(arr[0])
        for a in arr:
            a += [0 for i in range(LEN)]
        # 열의 수가 행의 수보다 많은 경우
    elif len(arr) < len(arr[0]):
        # 행의 수가 열의 수보다 많은 경우
        for i in range(len(arr[0]) - len(arr)):
            arr.append([0 for i in range(len(arr[0]))])

    return arr
